sgd/sgd-aa-oas3.yml

```
@@ -0,0 +1,577 @@
+    get:
+      operationId: get-version
+      summary: Returns the document describing this version of the API.
+      responses:
+        "200":
+          description: Ok
+          content:
+            text/html:
+              schema:
+                type: string
+        "404":
+          $ref: "#/components/responses/404"
+        "500":
+          $ref: "#/components/responses/500"
+      tags:
+        - public
+      x-spid-operation: # See §2.8
+        consentRequired: "N"
+        offlineAccess: true
+        offlineAccessDurationMax: 8640000
+
+  /tos:
+    get:
+      description: fdfd
+      operationId: get-tos
+      summary:
+        Returns the document with the terms and conditions related to this
+        system
+      responses:
+        "200":
+          description: Ok
+          content:
+            text/html:
+              schema:
+                type: string
+                description: HTML page with the Terms of Service
+        "404":
+          $ref: "#/components/responses/404"
+      tags:
+        - public
+      x-spid-operation: # See §2.8
+        consentRequired: "N"
+        offlineAccess: true
+        offlineAccessDurationMax: 8640000
+
+  /delegation/list:
+    get:
+      operationId: get-delegation
+      summary:
+        Returns the list of the delegations related to the services (or class of
+        services) provided by the authenticated RP and having the current user
+        as delegator
+      security:
+        - dpopToken: []
+          dpopProof: []
+      parameters:
+        - name: DPoP
+          in: header
+          required: true
+          description: DPoP proof
+          schema:
+            "$ref": "#/components/schemas/DPoP"
+      x-spid-operation: # See §2.8
+        consentRequired: "SP"
+        spidLevel: SPID2
+        offlineAccess: false
+        offlineAccessDurationMax: 0
+      tags:
+        - protected
+      responses:
+        "200":
+          description: Ok
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/DelegationList"
+              example:
+                delegation_list:
+                  - id: 1A00001AA21
+                    https://attributes.eid.gov.it/fiscal_number: TINIT-RSSMR*********I
+                  - id: 1A00002AA21
+                    https://attributes.eid.gov.it/fiscal_number: TINIT-ABCXX*********Z
+                  - id: 1A00004AA21
+                    https://attributes.eid.gov.it/fiscal_number: TINIT-DEFYY*********W
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+          #The requested service (or class of services) does not exist
+        "429":
+          $ref: "#/components/responses/429"
+        "500":
+          $ref: "#/components/responses/500"
+
+  /delegation/list/{service}:
+    get:
+      description: fdfd
+      operationId: get-delgation-service
+      summary:
```

```suggestion
  /services/{service}/delegations:
    get:
      description: fdfd
      operationId: list-delegation-for-service
      summary:
```

Valutare anche una ricerca

---

Non mi è molto chiara la tua proposta. Questo endpoint, di fatto, aggiunge a /delegation/list una funzionalità di filtro per servizi. Ad esempio dando in input un particolare identificativo di un servizio ottengo solo le deleghe che hanno quel servizio come oggetto.